### ISSUE TITLE
PHP unit tests failing

### ISSUE THREAD

**Issue by phil-davis** [owncloud/core#40098]
https://drone.owncloud.com/owncloud/core/35824/16/7
```
PHPUnit 9.5.20 #StandWithUkraine
Runtime:       PHPDBG 7.4.29
```
10622 tests - test run passed a few hours ago when merging #40092 

But now, PR #40097 - same PHP and PHPunit versions:
https://drone.owncloud.com/owncloud/core/35832/21/7
```
PHPUnit 9.5.20 #StandWithUkraine
Runtime:       PHPDBG 7.4.29
```

10622 tests - the same number - but:
```
There were 8 failures:

1) OCA\DAV\Tests\unit\Connector\Sabre\FileTest::testPutSingleFileLegalMtime with data set "long int" (9223372036854775807, 9223372036854775807)
Failed asserting that 15032385535 matches expected 9223372036854775807.

/drone/src/apps/dav/tests/unit/Connector/Sabre/FileTest.php:604
phpvfscomposer:///drone/src/lib/composer/phpunit/phpunit/phpunit:97
```

"something happened" (tm), but what?

**Comment by phil-davis** [comment]
They pass locally:

```
$ make test-php-unit NOCOVERAGE=true TEST_PHP_SUITE=apps/dav/tests/unit/Connector/Sabre/FileTest.php
php /usr/local/bin/composer install
Installing dependencies from lock file (including require-dev)
Verifying lock file contents can be installed on current platform.
Nothing to install, update or remove
Package container-interop/container-interop is abandoned, you should avoid using it. Use psr/container instead.
Package guzzlehttp/ringphp is abandoned, you should avoid using it. No replacement was suggested.
Package guzzlehttp/streams is abandoned, you should avoid using it. No replacement was suggested.
Package patchwork/jsqueeze is abandoned, you should avoid using it. No replacement was suggested.
Package swiftmailer/swiftmailer is abandoned, you should avoid using it. Use symfony/mailer instead.
Package jakub-onderka/php-console-color is abandoned, you should avoid using it. Use php-parallel-lint/php-console-color instead.
Package jakub-onderka/php-console-highlighter is abandoned, you should avoid using it. Use php-parallel-lint/php-console-highlighter instead.
Generating optimized autoload files
62 packages you are using are looking for funding.
Use the `composer fund` command to find out more!
Composer cleaner: Removed 5 files or directories.
PHPUNIT="/home/phil/git/owncloud/core/lib/composer/phpunit/phpunit/phpunit" build/autotest.sh sqlite apps/dav/tests/unit/Connector/Sabre/FileTest.php
Using PHP executable /usr/bin/php
Parsing all files in lib/public for the presence of @since or @deprecated on each method...

Using database oc_autotest
Setup environment for sqlite testing on local storage ...
Updated 0 paths from the index
Installing ....
creating sqlite db
ownCloud was successfully installed
Testing with sqlite ...
No coverage
/home/phil/git/owncloud/core/lib/composer/phpunit/phpunit/phpunit --configuration phpunit-autotest.xml --log-junit autotest-results-sqlite.xml ../apps/dav/tests/unit/Connector/Sabre/FileTest.php 
PHPUnit 9.5.20 #StandWithUkraine

Runtime:       PHP 7.4.29
Configuration: phpunit-autotest.xml

................................................................. 65 / 89 ( 73%)
........................                                          89 / 89 (100%)

Time: 00:05.628, Memory: 30.00 MB

OK (89 tests, 479 assertions)
```

**Comment by phil-davis** [comment]
Possibly related: https://stackoverflow.com/questions/47481402/seeking-in-very-large-files-on-32-bit-systems

"something is happening" that is causing the expected `PHP_INT_MAX` `9223372036854775807` not to be returned/possible on the system configuration in drone CI.

**Comment by phil-davis** [comment]
Some information on ext4 timestamp formats: https://www.sans.org/blog/understanding-ext4-part-2-timestamps/

There is a "strange" limit of `0x37fffffff` seconds from 1970-01-01 - `0x7fffffff` is the signed 32-bit integer part, an the 3 is the extra 2 bits that are available in the "extra" area of the timestamp storage in the inode structure.

**Comment by phil-davis** [comment]
Here is the full set of test fails, for reference:
```
There were 8 failures:

1) OCA\DAV\Tests\unit\Connector\Sabre\FileTest::testPutSingleFileLegalMtime with data set "long int" (9223372036854775807, 9223372036854775807)
Failed asserting that 15032385535 matches expected 9223372036854775807.

/drone/src/apps/dav/tests/unit/Connector/Sabre/FileTest.php:604
phpvfscomposer:///drone/src/lib/composer/phpunit/phpunit/phpunit:97

2) OCA\DAV\Tests\unit\Connector\Sabre\FileTest::testPutSingleFileLegalMtime with data set "too long int" (9.223372036854776E+18, 9223372036854775807)
Failed asserting that 15032385535 matches expected 9223372036854775807.

/drone/src/apps/dav/tests/unit/Connector/Sabre/FileTest.php:604
phpvfscomposer:///drone/src/lib/composer/phpunit/phpunit/phpunit:97

3) OCA\DAV\Tests\unit\Connector\Sabre\FileTest::testPutSingleFileLegalMtime with data set "long negative int" (-9223372036854775807, -9223372036854775807)
Failed asserting that -2147483648 matches expected -9223372036854775807.

/drone/src/apps/dav/tests/unit/Connector/Sabre/FileTest.php:604
phpvfscomposer:///drone/src/lib/composer/phpunit/phpunit/phpunit:97

4) OCA\DAV\Tests\unit\Connector\Sabre\FileTest::testPutSingleFileLegalMtime with data set "too long negative int" (-9223372036854775807-1, -9223372036854775807)
Failed asserting that -2147483648 matches expected -9223372036854775807.

/drone/src/apps/dav/tests/unit/Connector/Sabre/FileTest.php:604
phpvfscomposer:///drone/src/lib/composer/phpunit/phpunit/phpunit:97


5) OCA\DAV\Tests\unit\Connector\Sabre\FileTest::testChunkedPutLegalMtime with data set "long int" (9223372036854775807, 9223372036854775807)
Failed asserting that 15032385535 matches expected 9223372036854775807.

/drone/src/apps/dav/tests/unit/Connector/Sabre/FileTest.php:647
phpvfscomposer:///drone/src/lib/composer/phpunit/phpunit/phpunit:97

6) OCA\DAV\Tests\unit\Connector\Sabre\FileTest::testChunkedPutLegalMtime with data set "too long int" (9.223372036854776E+18, 9223372036854775807)
Failed asserting that 15032385535 matches expected 9223372036854775807.

/drone/src/apps/dav/tests/unit/Connector/Sabre/FileTest.php:647
phpvfscomposer:///drone/src/lib/composer/phpunit/phpunit/phpunit:97

7) OCA\DAV\Tests\unit\Connector\Sabre\FileTest::testChunkedPutLegalMtime with data set "long negative int" (-9223372036854775807, -9223372036854775807)
Failed asserting that -2147483648 matches expected -9223372036854775807.

/drone/src/apps/dav/tests/unit/Connector/Sabre/FileTest.php:647
phpvfscomposer:///drone/src/lib/composer/phpunit/phpunit/phpunit:97

8) OCA\DAV\Tests\unit\Connector\Sabre\FileTest::testChunkedPutLegalMtime with data set "too long negative int" (-9223372036854775807-1, -9223372036854775807)
Failed asserting that -2147483648 matches expected -9223372036854775807.

/drone/src/apps/dav/tests/unit/Connector/Sabre/FileTest.php:647
phpvfscomposer:///drone/src/lib/composer/phpunit/phpunit/phpunit:97
```

**Comment by phil-davis** [comment]
Reference about ext4 file system time stamps:
https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?id=a4dad1ae24f850410c4e60f22823cba1289b8d52

```
+/*
+ * We use an encoding that preserves the times for extra epoch "00":
+ *
+ * extra  msb of                         adjust for signed
+ * epoch  32-bit                         32-bit tv_sec to
+ * bits   time    decoded 64-bit tv_sec  64-bit tv_sec      valid time range
+ * 0 0    1    -0x80000000..-0x00000001  0x000000000 1901-12-13..1969-12-31
+ * 0 0    0    0x000000000..0x07fffffff  0x000000000 1970-01-01..2038-01-19
+ * 0 1    1    0x080000000..0x0ffffffff  0x100000000 2038-01-19..2106-02-07
+ * 0 1    0    0x100000000..0x17fffffff  0x100000000 2106-02-07..2174-02-25
+ * 1 0    1    0x180000000..0x1ffffffff  0x200000000 2174-02-25..2242-03-16
+ * 1 0    0    0x200000000..0x27fffffff  0x200000000 2242-03-16..2310-04-04
+ * 1 1    1    0x280000000..0x2ffffffff  0x300000000 2310-04-04..2378-04-22
+ * 1 1    0    0x300000000..0x37fffffff  0x300000000 2378-04-22..2446-05-10
+ *
+ * Note that previous versions of the kernel on 64-bit systems would
+ * incorrectly use extra epoch bits 1,1 for dates between 1901 and
+ * 1970.  e2fsck will correct this, assuming that it is run on the
+ * affected filesystem before 2242.
+ */
```

The summary is that a "modern file system" is likely to support future timestamps up to `0x37fffffff` which is a time on date 2446-05-10, and past timestamps with negative values back to `-0x80000000` which is a time on date 1901-12-13.

There seems not much point having unit tests that try to set file times outside of these ranges. The fact that they have been passing seems to be some file-system-data-caching "feature". It seems that some in-memory structures for these timestamps are implemented as full 64-bit integers. But the actual storage locations in inode structures on-disk that write these to persistent storage can only fit values withing the date ranges above.

**Comment by phil-davis** [comment]
Fixed by #40101
